content/ADR-145-pick-realm-v2.md

```diff
@@ -3,49 +3,69 @@ layout: adr
+- **FORCE_CATALYST**: This new link will enable the forcible selection of a Catalyst listed in its configuration.
+
+Furthermore, four new variants will be added, each containing a set of rules designed to address specific contexts.
+
+### Server-side
+
+This algorithm will be implemented on the server-side to accommodate various client implementations. While this may require modifying the implementation of the `LARGE_LATENCY` rule to receive necessary information from the client-side, it will enable enhancements such as adding a cache mechanism or event-driven notifications to proactively filter out unhealthy Catalysts, thereby improving the overall performance of the algorithm.
+
+To fulfill the goal of creating a cost-effective, scalable and globally accessible service, two implementation proposals, CloudFlare Workers and AWS Lambdas, are being considered. The optimal choice for this service will be determined by analyzing various aspects related to cost, pricing model, performance, and latency, which will be discussed in detail in the Server-Side Implementation section.
+
+To assess these aspects, we will examine the workload of the current algorithm, which is monitored through Segment events. The chart below presents the workload data for the past 4 months as of the time of writing.
+
+<img src="resources/ADR-145/realm-picking-algorithm-workload.png" alt="drawing" style="width:100%;"/>
+
+_The average workload is 10k realm picks per day._
 
 ## Specification
 
-### Accepting Users Filter
+### Overloaded Catalyst
 
 _Description_:
 
-Publicly exposing the Catalyst Node's available resources is a security risk, so their estimation should be done in the Catalyst (specifically in the BFF) and not in the Kernel.
+Exposing the available resources of Catalyst Nodes publicly poses a security risk, and thus, their estimation should be conducted within the Catalyst itself (_specifically in the BFF_), rather than in the Kernel.
 
-This way, the BFF will leverage the usage of the field `accepting_users` in the `/about` endpoint.
+Consequently, the BFF will utilize the `acceptingUsers` field in the `/about` endpoint to leverage the calculation of this value. The logic on the Kernel side will be straightforward, involving filtering out Catalysts with a value of false for this field.
 
-The logic from the Kernel side will be simple: it will filter out the catalysts having that value as false.
-
-On the other hand, the BFF will have a set of rules to calculate that value. The `accepting_users` field will be false if:
-- The max amount of users for the server is reached
+On the other hand, the BFF will implement a set of rules to determine the value of `acceptingUsers`. The `acceptingUsers` field will be set to false if:
+- The maximum number of users for the server has been reached, with the maximum amount being specified through environment variables.
 - The CPU is at 90% or higher
 - The memory usage is at 90% or higher
 
```

Comment:
The pseudo-code below is no longer valid, as that implementation lives in Kernel. Please remove it

Reply:
All the pseudo-codes were removed until we have the final implementation in the proposed service. I'll update this ADR to add the pseudocode for each rule once we have each implementation in place to avoid re-work.

Done here -> [feedback: remove pseudo-codes](https://github.com/decentraland/adr/pull/220/commits/1891f6ba8532b35fc25585c718f749f4f523bff1)